### PR TITLE
Migrate github checks to buildkite for metricbeat and x-pack/metricbeat

### PR DESCRIPTION
**Pull request by ev1yehor**
## Proposed commit message
Migrate github checks to buildkite for metricbeat and x-pack/metricbeatx

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- https://github.com/elastic/ingest-dev/issues/3380

### CODE REVIEW

File: .buildkite/metricbeat/pipeline.yml [hunk 1]
```diff
@@ -30,6 +30,32 @@ env:
   TEST_COVERAGE: "true"
 
 steps:
+  - group: "Metricbeat Check/Update"
+    key: "metricbeat-check-update"
+    steps:
+      - label: "Run check/update"
+        command: |
+          make -C metricbeat check update
```
Comment (dliappis) [inline .buildkite/metricbeat/pipeline.yml:38]: I wonder, if this fails and the next command succeeds (assuming it's possible) the job would be successful right?
So, I think we want to fail fast, i.e. either add `set -eo pipefail` before the two `make` commands, or, instead:

`make -C metricbeat check update && make check-no-changes`

what do you think?

Comment (ev1yehor) [reply .buildkite/metricbeat/pipeline.yml:38]: Good point! How do you think, will be better to do it in separate PR for all beats?

Comment (ev1yehor) [reply .buildkite/metricbeat/pipeline.yml:38]: Change will be done for all beats in separate PR.

**Review by dliappis — APPROVED** [review]
LGTM

**Pull request merged**